bindings/rust/bench/src/bin/graph_perf.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use csv::Reader;
+use plotters::{
+    prelude::{
+        BindKeyPoints, ChartBuilder, ErrorBar, IntoDrawingArea, LabelAreaPosition, Rectangle,
+        SVGBackend,
+    },
+    series::LineSeries,
+    style::{AsRelative, Color, IntoFont, Palette, Palette99, RGBAColor, BLACK, WHITE},
+};
+use semver::Version;
+use std::{collections::HashMap, error::Error};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let mut csv_reader = Reader::from_path("historical-perf/perf.csv")?;
+
+    // map bench name -> points (x, y, y_bar)
+    let mut map: HashMap<&str, Vec<(Version, f64, f64)>> = HashMap::new();
+
+    // get headers of csv as Vec<String>
+    let headers = csv_reader.headers()?.clone();
+    let headers: Vec<String> = headers
+        .iter()
+        .skip(1) // skip first header "tag"
+        .step_by(2) // skip stderr headers
+        .map(|s| s.to_string())
+        .collect::<Vec<String>>();
+
+    // initialize data vectors
+    for header in headers.iter() {
+        map.insert(header, Vec::new());
+    }
+
+    // read in all the tags as Versions
+    let mut tag_names = Reader::from_path("historical-perf/perf.csv")?
+        .records()
+        .map(|record_res| Version::parse(&record_res.unwrap()[0][1..]).unwrap())
+        .collect::<Vec<_>>();
+    let num_tags = tag_names.len();
+
+    // fill in missing versions (1.3.15, 1.3.30-1.3.37)
+    tag_names.push(Version::new(1, 3, 15));
```

Comment:
The tag 1.3.15 is to make the graph prettier so that the first point doesn't start exactly on the y-axis, and v1.3.30-v1.3.37 are missing because of dependency issues while generating the rust bindings. I could figure them out, but then that would add a lot of complexity in the code to deal with those specific versions for not that much gain, especially since we can see performance before and after that up to a year ago. There aren't any points graphed for those versions, so I include them in the x-axis purely so that the scale of the x-axis stays constant at one version per tick; otherwise, it wouldn't be immediately obvious that v1.3.29 jumps straight to v1.3.38. Again, I think these things are good to callout in the readme, which I'll work on.

Reply:
To make it two separate lines, we would need to plot two `LineSeries`, which would increase the complexity of the code a lot, especially given the plotting code is already pretty complicated because of the `plotters` library. I think the current chart format with no points are plotted on the line between v1.3.29 to v.1.3.38 is good enough for the users of the benching suite, mainly us as developers to make sure no drastic performance declines are happening with future versions.